File: API/Entities/Member.cs

using System.ComponentModel.DataAnnotations.Schema;
using System.Data;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{
    public string Id { get; set; } = null!;

    public DateOnly DateofBirth { get; set; }

    public string? ImageUrl { get; set; }

    public string? Email { get; set;     }

    public required string DisplayName { get; set; }

    public DateTime Created { get; set; } = DateTime.UtcNow;

    public DateTime LastActive { get; set; } = DateTime.UtcNow;

    public required string Gender { get; set; }

    public required string City { get; set; }

    public required string Country { get; set; }

    public string? Description { get; set; }

    [JsonIgnore]
    public List<Photo> Photos { get; set; } = [];
    

    [JsonIgnore]
    [ForeignKey(nameof(Id))]
    public AppUser User { get; set; } = null!;
   
}

File: client/src/app/types/Member.ts
export type Member = {
  id: string
  dateofBirth: string
  imageUrl?: string
  email: any
  displayName: string
  created: string
  lastActive: string
  gender: string
  city: string
  country: string
  description?: string

}

export type Photo = {
  id: number
  url: string
  publicId: any
  member: any
  memberId: string
}

export type editablemember = {
  displayName : string;
  description? : string;
  city: string;
  country: string;  
}

export class MemberParams
{
  gender? : string;
  minAge = 18;
  maxAge = 100;
  pageNumber = 1;
  pageSize = 10;
  orderBy = 'lastActive';

}